zerver/tests/test_users.py

```diff
@@ -2545,6 +2556,47 @@ def test_restricted_access_to_users(self) -> None:
         inaccessible_user_ids = [user["user_id"] for user in inaccessible_users]
         self.assertCountEqual(inaccessible_user_ids, [cordelia.id, desdemona.id, othello.id])
 
+    def test_get_user_with_restricted_access(self) -> None:
+        polonius = self.example_user("polonius")
+        hamlet = self.example_user("hamlet")
+        othello = self.example_user("othello")
+        iago = self.example_user("iago")
+        prospero = self.example_user("prospero")
+        aaron = self.example_user("aaron")
+        zoe = self.example_user("ZOE")
+        shiva = self.example_user("shiva")
+        cordelia = self.example_user("cordelia")
+        desdemona = self.example_user("desdemona")
+        realm = get_realm("zulip")
+        reset_email_visibility_to_everyone_in_zulip_realm()
+
+        self.set_up_db_for_testing_user_access()
+
+        members_group = UserGroup.objects.get(name="role:members", realm=realm)
+        do_change_realm_permission_group_setting(
+            realm, "can_access_all_users_group", members_group, acting_user=None
+        )
```

Comment:
I think these two lines are already present in `set_up_db_for_testing_user_access`

Reply:
Removed those lines.